dali/pipeline/operator/operator.h

```diff
@@ -152,7 +152,39 @@ class DLL_PUBLIC OperatorBase {
    * @brief For reader Ops, returns the size of the dataset
    * For all other Ops, returns -1
    */
-  DLL_PUBLIC virtual Index epoch_size() const {
+  DLL_PUBLIC virtual Index epoch_size(bool consider_padding = true) const {
+    return -1;
+  }
+
+  /**
+   * @brief For reader Ops, returns the number of shards for given reader
+   * For all other Ops, returns -1
+   */
+  DLL_PUBLIC virtual int number_of_shards() const {
+    return -1;
+  }
+
+  /**
+   * @brief For reader Ops, returns the shard id
+   * For all other Ops, returns -1
+   */
+  DLL_PUBLIC virtual int shard_id() const {
+    return -1;
+  }
+
+  /**
+   * @brief For reader Ops, returns if padding is applied in the reader
+   * For all other Ops, returns -1
+   */
+  DLL_PUBLIC virtual int pad_last_batch() const {
+    return -1;
+  }
+
+  /**
+   * @brief For reader Ops, returns if the reader should stick to the assigned shard
+   * For all other Ops, returns -1
+   */
+  DLL_PUBLIC virtual int stick_to_shard() const {
     return -1;
   }
```

Comment:
I was thinking, maybe instead of whole multitude of functions, maybe we should have something like `GetReaderMetadata()`? 

Btw. I think we all want to switch to snake_case_for_cpp_functions.

Reply:
Done